packages/medusa/src/services/product-category.ts

```diff
@@ -0,0 +1,67 @@
+import { isDefined, MedusaError } from "medusa-core-utils"
+import { EntityManager } from "typeorm"
+import { TransactionBaseService } from "../interfaces"
+import { ProductCategory } from "../models"
+import { ProductCategoryRepository } from "../repositories/product-category"
+import { FindConfig, Selector } from "../types/common"
+import { buildQuery } from "../utils"
+
+type InjectedDependencies = {
+  manager: EntityManager
+  productCategoryRepository: typeof ProductCategoryRepository
+}
+
+/**
+ * Provides layer to manipulate product categories.
+ */
+class ProductCategoryService extends TransactionBaseService {
+  protected manager_: EntityManager
+  protected readonly productCategoryRepository_: typeof ProductCategoryRepository
+  protected transactionManager_: EntityManager | undefined
+
+  constructor({
+    manager,
+    productCategoryRepository,
+  }: InjectedDependencies) {
+    super(arguments[0])
+    this.manager_ = manager
+
+    this.productCategoryRepository_ = productCategoryRepository
+  }
+
+  /**
+   * Retrieves a product category by id.
+   * @param id - the id of the product category to retrieve.
+   * @param config - the config of the product category to retrieve.
+   * @return the product category.
+   */
+  async retrieve(
+    id: string,
```

Comment:
todo(nit): If the id is undefined, we will respond with `id must be defined`. It is not immediately obvious to the consumer, that it was thrown in the product category service, so I suggest renaming it. 
```suggestion
    productCategoryId: string,
```